packages/manifold/contracts/gachaclaims/ERC1155GachaLazyClaim.sol

```diff
@@ -0,0 +1,256 @@
+    uint256 instanceId,
+    UpdateClaimParameters memory updateClaimParameters
+  ) external override adminRequired {
+    Claim memory claim = _getClaim(creatorContractAddress, instanceId);
+    if (updateClaimParameters.endDate != 0 && updateClaimParameters.startDate >= updateClaimParameters.endDate)
+      revert IGachaLazyClaim.InvalidDate();
+    if (updateClaimParameters.totalMax < claim.total) revert IGachaLazyClaim.CannotLowerTotalMaxBeyondTotal();
+    if (updateClaimParameters.storageProtocol == StorageProtocol.INVALID) revert IGachaLazyClaim.InvalidStorageProtocol();
+
+    // Overwrite the existing values
+    _claims[creatorContractAddress][instanceId] = Claim({
+      storageProtocol: claim.storageProtocol,
+      total: claim.total,
+      totalMax: updateClaimParameters.totalMax,
+      startDate: updateClaimParameters.startDate,
+      endDate: updateClaimParameters.endDate,
+      startingTokenId: claim.startingTokenId,
+      tokenVariations: claim.tokenVariations,
+      location: updateClaimParameters.location,
+      paymentReceiver: updateClaimParameters.paymentReceiver,
+      cost: updateClaimParameters.cost,
+      erc20: claim.erc20
+    });
+
+    // logic for updating tokenIds
+    emit GachaClaimUpdated(creatorContractAddress, instanceId);
+  }
+
+  /**
+   * See {IERC1155GachaLazyClaim-getClaim}.
+   */
+  function getClaim(address creatorContractAddress, uint256 instanceId) public view override returns (Claim memory) {
+    return _getClaim(creatorContractAddress, instanceId);
+  }
+
+  /**
+   * See {IERC1155GachaLazyClaim-getClaimForToken}.
+   */
+  function getClaimForToken(
+    address creatorContractAddress,
+    uint256 tokenId
+  ) external view override returns (uint256 instanceId, Claim memory claim) {
+    instanceId = _tokenInstances[creatorContractAddress][tokenId];
+    claim = _getClaim(creatorContractAddress, instanceId);
+  }
+
+  function _getClaim(address creatorContractAddress, uint256 instanceId) private view returns (Claim storage claim) {
+    claim = _claims[creatorContractAddress][instanceId];
+    if (claim.storageProtocol == StorageProtocol.INVALID) revert IGachaLazyClaim.ClaimNotInitialized();
+  }
+
+  /**
+   * See {IGachaLazyClaim-mintReserve}.
+   */
+  function mintReserve(address creatorContractAddress, uint256 instanceId, uint32 mintCount) external payable override {
+    if (Address.isContract(msg.sender)) revert IGachaLazyClaim.CannotMintFromContract();
+    Claim storage claim = _getClaim(creatorContractAddress, instanceId);
+    // Checks for reserving
+    if (claim.startDate > block.timestamp) revert IGachaLazyClaim.ClaimInactive();
+    if (claim.endDate > 0 && claim.endDate <= block.timestamp) revert IGachaLazyClaim.ClaimInactive();
+    if (msg.value != (claim.cost + MINT_FEE) * mintCount) revert IGachaLazyClaim.InvalidPayment();
+    uint32 amountAvailable = claim.totalMax - claim.total;
+    if (amountAvailable == 0) revert IGachaLazyClaim.ClaimSoldOut();
```

Comment:
ran into an edge case here! if claim.totalMax is 0, and on a first mint, where claim.total = 0, the amount available is still 0 (and for claim.totalMax = 0, it might underflow in future mints)

logic here should prolly be

if (claim.totalMax != 0) // check available amounts?